### DISCUSSION TITLE
How to modify this alliance chain into a public chain？

### DISCUSSION THREAD

**Discussion by yangmiok** [ConsenSys/quorum#1640] [General]
if tally := snap.Tally[candidate]; tally.Votes > snap.ValSet.Size()/2 {
        //    if tally := snap.Tally[candidate]; tally.Votes > 0 {
        if tally.Authorize {
            logger.Info("BFT: reached majority to add validator")
            snap.ValSet.AddValidator(candidate)
        } else {
            logger.Info("BFT: reached majority to remove validator")
            snap.ValSet.RemoveValidator(candidate)

            // Discard any previous votes the deauthorized validator cast
            for i := 0; i < len(snap.Votes); i++ {
                if snap.Votes[i].Validator == candidate {
                    // Uncast the vote from the cached tally
                    snap.uncast(snap.Votes[i].Address, snap.Votes[i].Authorize)

                    // Uncast the vote from the chronological list
                    snap.Votes = append(snap.Votes[:i], snap.Votes[i+1:]...)

                    i--
                }
            }
        }
I try to change tally.Votes > 0 to become a public chain，but fail？how

**Comment by baptiste-b-pegasys** [comment]
what is failing exactly?
